src/context-parser.js

```diff
@@ -180,6 +180,20 @@ FastParser.prototype.walk = function(i, input, endsWithEOF) {
             if(this.tags[0].toLowerCase() === this.tags[1].toLowerCase()) {
                 reconsume = 0;  /* see 12.2.4.13 - switch state for the following case, otherwise, reconsume. */
                 this.matchEndTagWithStartTag(symbol);
+                
+                /* 
+                The reason we reset tags[0] (start tag) is that there is no trivial way to reset RCDATA/RAWTEXT/SCRIPT state to DATA state.
+                By setting tags[0] = '', we would expect processTagName() to transit the state back the DATA state
```

Comment:
so why there is any switch case in the function processTagName to set the state back to DATA? am i missing something?